File: src/SnipSnap/Models/CaptureResult.cs
using System.Drawing;
using System.Windows.Media.Imaging;

namespace SnipSnap.Models;

public class CaptureResult
{
    public CaptureMode Mode { get; init; }
    public BitmapSource? Screenshot { get; init; }
    public string? VideoPath { get; init; }
    public DateTime CapturedAt { get; init; } = DateTime.Now;
    public System.Drawing.Rectangle CaptureRegion { get; init; }

    public bool IsVideo => Mode.IsVideo();
    public bool IsScreenshot => Mode.IsScreenshot();
    public bool HasContent => Screenshot != null || !string.IsNullOrEmpty(VideoPath);
}

File: snipr-linux/snipr/models/capture_result.py
from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from gi.repository import GdkPixbuf

from snipr.models.capture_mode import CaptureMode


@dataclass
class CaptureResult:
    mode: CaptureMode
    screenshot: GdkPixbuf.Pixbuf | None = None
    video_path: str | None = None
    captured_at: datetime = field(default_factory=datetime.now)
    capture_region: tuple[int, int, int, int] = (0, 0, 0, 0)  # x, y, w, h

    @property
    def is_video(self) -> bool:
        return self.mode.is_video

    @property
    def is_screenshot(self) -> bool:
        return self.mode.is_screenshot

    @property
    def has_content(self) -> bool:
        return self.screenshot is not None or self.video_path is not None
